rotkehlchen/chain/evm/decoding/curve/decoder.py

```diff
@@ -547,6 +570,71 @@ def _decode_curve_trades(self, context: DecoderContext) -> DecodingOutput:
 
         return DEFAULT_DECODING_OUTPUT
 
+    def _pre_decode_deposit_and_stake(self, context: DecoderContext) -> DecodingOutput:
+        """
+        Enrich the transfer for deposit and stake to save the amount of gauge tokens
+        received. We need to match against transfers manually because they are not
+        made from the user address but an intermediate contract
+        """
+        if (pool_assets := self.pools.get(context.tx_log.address)) is None:
+            log.error(
+                f'Curve pool for {context.tx_log.address} not present in cache at '
+                f'{context.transaction.tx_hash.hex()}. Skipping',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        provider = hex_or_bytes_to_address(context.tx_log.topics[1])
+        deposited_amounts = [
+            hex_or_bytes_to_int(deposit_amount)
+            for deposit_amount in [context.tx_log.data[i:i + 32] for i in range(len(pool_assets))]
+        ]
+        resolved_assets = [
+            Asset(evm_address_to_identifier(
+                address=address,
+                chain_id=self.evm_inquirer.chain_id,
+                token_type=EvmTokenKind.ERC20,
+            )) if address != ETH_SPECIAL_ADDRESS else A_ETH for address in pool_assets
+        ]
+
+        gauge_tokens = None
+        for event in context.decoded_events:
+            if event.asset not in resolved_assets:
+                continue
+
+            # The context.tx_log is a deposit event from curve. We deposit a underlying token
+            # of the pool (for example EURE) and we get an amount of the pool token. The problem
+            # is that the event log doesn't contain the amount of the LP received and it is needed
+            # later when decoding the gauge events since the amounts that we compare with are
+            # different. The next for loop iterates over the logs to find the transfer that follows
+            # the current log entry and extract from there the amount of LP tokens obtained after
+            # the deposit.
+            for tx_log in context.all_logs:
+                if tx_log.log_index > context.tx_log.log_index:
+                    if (
+                        tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                        hex_or_bytes_to_address(tx_log.topics[1]) == provider
+                    ):
+                        gauge_tokens = hex_or_bytes_to_int(tx_log.data[0:32])
```

Comment:
Why not combine those? Weird that the linter did not complain for this .. unless I am missing something .. or wait is the break on the first if? So you can do `gauge_tokens = hex_or_bytes_to_int(tx_log.data[0:32])` multiple times and only take the last one? Why? This needs explanation.

```suggestion
                if (
                    tx_log.log_index > context.tx_log.log_index and
                    tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
                    hex_or_bytes_to_address(tx_log.topics[1]) == provider
                ):
                    gauge_tokens = hex_or_bytes_to_int(tx_log.data[0:32])
```